src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,959 @@
+        {
+            get
+            {
+                ushort signal =
+                    ReadUInt16((ushort)Registers.VL53L1X_RESULT__PEAK_SIGNAL_COUNT_RATE_CROSSTALK_CORRECTED_MCPS_SD0);
+                ushort spNb = ReadUInt16((ushort)Registers.VL53L1X_RESULT__DSS_ACTUAL_EFFECTIVE_SPADS_SD0);
+                return (ushort)(2000.0 * signal / spNb);
+            }
+        }
+
+        /// <summary>
+        /// Returns the ambient per SPAD in kcps/SPAD.
+        /// </summary>
+        public ushort AmbientPerSpad
+        {
+            get
+            {
+                ushort ambientRate = ReadUInt16((ushort)Registers.RESULT__AMBIENT_COUNT_RATE_MCPS_SD);
+                ushort spNb = ReadUInt16((ushort)Registers.VL53L1X_RESULT__DSS_ACTUAL_EFFECTIVE_SPADS_SD0);
+                return (ushort)(2000.0 * ambientRate / spNb);
+            }
+        }
+
+        /// <summary>
+        /// Returns the signal in kcps.
+        /// </summary>
+        public ushort SignalRate
+        {
+            get
+            {
+                return (ushort)(ReadUInt16((ushort)Registers
+                    .VL53L1X_RESULT__PEAK_SIGNAL_COUNT_RATE_CROSSTALK_CORRECTED_MCPS_SD0) * 8);
+            }
+        }
+
+        /// <summary>
+        /// Returns the current number of enabled SPADs.
+        /// </summary>
+        public ushort SpadNb
+        {
+            get
+            {
+                return (ushort)(ReadUInt16((ushort)Registers.VL53L1X_RESULT__DSS_ACTUAL_EFFECTIVE_SPADS_SD0) >> 8);
+            }
+        }
+
+        /// <summary>
+        /// Returns the ambient rate in kcps.
+        /// </summary>
+        public ushort AmbientRate
+        {
+            get
+            {
+                return (ushort)(ReadUInt16((ushort)Registers.RESULT__AMBIENT_COUNT_RATE_MCPS_SD) * 8);
+            }
+        }
+
+        /// <summary>
+        /// Returns the <see cref="RangeStatus" /> of the device.
+        /// </summary>
+        public RangeStatus RangeStatus
+        {
+            get
+            {
+                byte rgSt = ReadByte((ushort)Registers.VL53L1X_RESULT__RANGE_STATUS);
+
+                rgSt = (byte)(rgSt & 0x1F);
+
+                switch (rgSt)
+                {
+                    case 9:
+                        return RangeStatus.NoError;
+                    case 6:
+                        return RangeStatus.SigmaFailure;
+                    case 4:
+                        return RangeStatus.SignalFailure;
+                    case 5:
+                        return RangeStatus.OutOfBounds;
+                    case 7:
+                        return RangeStatus.WrapAround;
+                    default:
+                        throw new Exception($"The returned range status code {rgSt} of the device is unknown.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the offset correction value in mm.
+        /// </summary>
+        public short Offset
```

Comment:
Same here, as it's a distance, we should use UnitsNet
Also, please give in the intellisence comments the typical range.

Reply:
Now using Length instead of ushort.
Unfortunately, the api description of ST does not mention any range for the Offset. All they are saying is that 0 is no offset which is kinda obvious...